catalog-rest-service/src/test/java/org/openmetadata/catalog/CatalogApplicationTest.java

```diff
@@ -13,55 +13,99 @@
 import io.dropwizard.testing.junit5.DropwizardAppExtension;
-import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
 import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
 import javax.ws.rs.client.WebTarget;
 import lombok.extern.slf4j.Slf4j;
+import org.flywaydb.core.Flyway;
 import org.glassfish.jersey.client.ClientProperties;
 import org.glassfish.jersey.client.HttpUrlConnectorProvider;
-import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.openmetadata.catalog.fernet.Fernet;
 import org.openmetadata.catalog.resources.CollectionRegistry;
-import org.openmetadata.catalog.resources.EmbeddedMySqlSupport;
 import org.openmetadata.catalog.resources.events.WebhookCallbackResource;
+import org.testcontainers.containers.JdbcDatabaseContainer;
 
 @Slf4j
-@ExtendWith(EmbeddedMySqlSupport.class)
-@ExtendWith(DropwizardExtensionsSupport.class)
 public abstract class CatalogApplicationTest {
-  protected static final String CONFIG_PATH;
-  public static final DropwizardAppExtension<CatalogApplicationConfig> APP;
-  private static final Client client;
+  protected static final String CONFIG_PATH = ResourceHelpers.resourceFilePath("openmetadata-secure-test.yaml");
+  private static JdbcDatabaseContainer<?> SQL_CONTAINER;
+  public static DropwizardAppExtension<CatalogApplicationConfig> APP;
   protected static final WebhookCallbackResource webhookCallbackResource = new WebhookCallbackResource();
   public static final String FERNET_KEY_1 = "ihZpp5gmmDvVsgoOG6OVivKWwC9vd5JQ";
-  public static final String FERNET_KEY_2 = "0cDdxg2rlodhcsjtmuFsOOvWpRRTW9ZJ";
 
   static {
     CollectionRegistry.addTestResource(webhookCallbackResource);
-    CONFIG_PATH = ResourceHelpers.resourceFilePath("openmetadata-secure-test.yaml");
-    APP = new DropwizardAppExtension<>(CatalogApplication.class, CONFIG_PATH);
-    client = ClientBuilder.newClient();
-    client.property(ClientProperties.CONNECT_TIMEOUT, 0);
-    client.property(ClientProperties.READ_TIMEOUT, 0);
-    client.property(HttpUrlConnectorProvider.SET_METHOD_WORKAROUND, true);
+
     Fernet.getInstance().setFernetKey(FERNET_KEY_1);
   }
 
-  public static WebTarget getResource(String collection) {
-    String targetURI = "http://localhost:" + APP.getLocalPort() + "/api/v1/" + collection;
-    return client.target(targetURI);
+  @BeforeAll
+  public static void createApplication() throws Exception {
+    // The system properties are provided by maven-surefire for testing with mysql and postgres
+    final String jdbcContainerClassName = System.getProperty("jdbcContainerClassName");
+    final String jdbcContainerImage = System.getProperty("jdbcContainerImage");
+
+    SQL_CONTAINER =
+        (JdbcDatabaseContainer<?>)
+            Class.forName(jdbcContainerClassName).getConstructor(String.class).newInstance(jdbcContainerImage);
+    SQL_CONTAINER.start();
+
```

Comment:
@mosiac1, can we try starting this only once for all the tests based on a flag? Will that work? We can cleanup the database using flyway every time.

Reply:
AFAIK there is no simple way to do this with junit5 since all test classes are treated independently